src/main/java/com/wanikani/api/v2/request/StudyMaterialsRequest.java

```diff
@@ -1,66 +1,24 @@
 package com.wanikani.api.v2.request;
 
-import com.wanikani.api.v2.model.SubjectType;
+import lombok.Singular;
+import lombok.experimental.SuperBuilder;
 
-import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.List;
-import java.util.stream.Collectors;
 
-import static com.wanikani.api.v2.request.QueryStringUtils.append;
-import static com.wanikani.api.v2.request.QueryStringUtils.appendList;
-
-public class StudyMaterialsRequest implements Request {
-
-    private final String queryString;
-
-    public StudyMaterialsRequest(String queryString) {
-        this.queryString = queryString;
-    }
-
-    public static Builder builder() {
-        return new Builder();
-    }
+@SuperBuilder(toBuilder = true)
+public class StudyMaterialsRequest extends CollectionRequest {
+    @Singular
+    private List<Long> subjectIds;
+    @Singular
+    private List<String> subjectTypes;
+    private Boolean hidden;
 
+    @Override
     public String getQueryString() {
-        return queryString;
-    }
-
-    public static class Builder extends CollectionBuilder<StudyMaterialsRequest, Builder> {
-        private Boolean hidden;
-        private List<Long> subjectIds = new ArrayList<>();
-        private List<String> subjectTypes = new ArrayList<>();
-
-        public Builder hidden(boolean hidden) {
-            this.hidden = hidden;
-            return this;
-        }
-
-        public Builder subjectIds(List<Long> subjectIds) {
-            this.subjectIds = subjectIds;
-            return this;
-        }
-
-        public Builder subjectIds(long... ids) {
-            return subjectIds(Arrays.stream(ids).boxed().collect(Collectors.toList()));
-        }
-
-        public Builder subjectTypes(List<SubjectType> subjectTypes) {
-            this.subjectTypes = subjectTypes.stream().map(SubjectType::getName).collect(Collectors.toList());
-            return this;
-        }
-
-        public Builder subjectTypes(SubjectType... subjectTypes) {
-            return subjectTypes(Arrays.stream(subjectTypes).collect(Collectors.toList()));
-        }
```

Comment:
It's absolutely possible, but because Lombok takes an existing builder method with the same identifier to mean "don't generate this", it involves redeclaring the List-accepting version as well as the vararg version for every List field we have.

It's easily doable if you think the vararg version is an enough of an improvement to bother - I think it _sort of_ is, but I also think what Singular generates is perfectly serviceable as well. Up to you!